estuary/ui
# UX Tweaks: bindings selector

## Changes

The following features are included in this PR:

1. Default the resource config of collections added to a materialization via a materialize CTA according to the collection name annotation.

1. Display the schema inference CTA in the _Output Collections_ form section when a collection schema contains the `x-infer-schema` annotation.

## Tests

The following tests were conducted:

* Create a capture using a _schema-inference enabled_ connector and confirm that the schema inference CTA remains hidden in the capture create workflow.

* Edit a capture that **only** has _schema-inference enabled_, bound collections and confirm that the schema inference CTA appears.

* Edit a capture that **only** has _schema-inference disabled_, bound collections and confirm that the schema inference CTA remains hidden.

* Edit a capture that has a mix of bound collections and confirm that the schema inference CTA appears **only** when a _schema-inference enabled_, bound collection is selected.

**NOTE:** The final three tests were repeated for the materialization create and edit workflows.

## Issues

The issues directly below are completely resolved by this PR:
#481

## Screenshots

### Default Resource Config Field of Prefilled Collections

**Materialization Edit | Original Task Bindings**

<img width="1920" alt="pr_screenshot-487-bindings_selector_ux_tweaks-prefill_collection-default" src="https://user-images.githubusercontent.com/77648584/220703022-b8515d70-e02e-43b8-9cac-0965069e4661.PNG">

<br />

**Materialization Edit | Prefilled Resource Config**

<img width="1920" alt="pr_screenshot-487-bindings_selector_ux_tweaks-prefill_collection-redux_dev_tools" src="https://user-images.githubusercontent.com/77648584/220703060-13280607-16fd-4621-9b38-8e4e54a13ccf.PNG">

### Schema Inference CTA

**Capture Edit | Schema Inference Enabled**

<img width="1920" alt="pr_screenshot-487-bindings_selector_ux_tweaks-infer-schema-enabled" src="https://user-images.githubusercontent.com/77648584/220703172-4ee05925-0dd1-44cd-8f39-f7370805d187.PNG">

<br />

**Capture Edit | Schema Inference Disabled**

<img width="1920" alt="pr_screenshot-487-bindings_selector_ux_tweaks-infer-schema-disabled" src="https://user-images.githubusercontent.com/77648584/220703200-1ac55c03-d6a3-44ba-9499-d893f6e1ccbc.PNG">

<br />
